worker/uniter/relation/relationer.go

```diff
@@ -53,15 +53,15 @@ func (r *Relationer) RelationUnit() *apiuniter.RelationUnit {
 
 // Join initializes local state and causes the unit to enter its relation
 // scope, allowing its counterpart units to detect its presence and settings
-// changes. Local state directory is not created until needed.
+// changes.
 func (r *Relationer) Join() error {
 	if r.dying {
 		panic("dying relationer must not join!")
```

Comment:
As a drive-by, can you replace this panic with an error?

Reply:
I'd rather not do this as a drive-by.  Will double check that the uniter will be restarted on an error here and address in a new PR

Reply:
Fair enough; we can definitely do this in a separate PR as you suggest.